pkg/monitor/cluster/clustersync.go

```diff
@@ -13,57 +14,54 @@ import (
 
 func (mon *Monitor) emitClusterSync(ctx context.Context) error {
 	if mon.hiveClusterManager == nil {
-		// TODO(hive): remove this once we have Hive everywhere.
+		// TODO(hive): remove this once we have HiveManager available everywhere
 		mon.log.Info("skipping: no hive cluster manager")
 		return nil
 	}
 
 	clusterSync, err := mon.hiveClusterManager.GetClusterSync(ctx, mon.doc)
 	if err != nil {
+		mon.log.Errorf("Error in getting the clustersync data: %v", err)
 		return err
 	}
-	if clusterSync == nil {
-		return nil
-	} else {
-		if clusterSync.Status.SyncSets == nil {
-			return nil
-		} else {
+	if clusterSync != nil {
+		if clusterSync.Status.SyncSets != nil {
 			for _, s := range clusterSync.Status.SyncSets {
 				mon.emitGauge("hive.clustersync", 1, map[string]string{
-					"syncType": "SyncSets",
-					"name":     s.Name,
-					"result":   string(s.Result),
+					"name":    s.Name,
+					"status":  string(s.Result),
+					"type":    "SyncSets",
+					"message": cleanString(s.FailureMessage),
 				})
 
 				if mon.hourlyRun {
 					mon.log.WithFields(logrus.Fields{
-						"syncType":           "SyncSets",
 						"name":               s.Name,
-						"result":             string(s.Result),
+						"status":             string(s.Result),
+						"type":               "SyncSets",
 						"firstSuccessTime":   timeToString(s.FirstSuccessTime),
 						"lastTransitionTime": timeToString(&s.LastTransitionTime),
-						"failureMessage":     s.FailureMessage,
+						"message":            cleanString(s.FailureMessage),
 					}).Print()
 				}
 			}
 		}
-		if clusterSync.Status.SelectorSyncSets == nil {
-			return nil
-		} else {
+		if clusterSync.Status.SelectorSyncSets != nil {
 			for _, s := range clusterSync.Status.SelectorSyncSets {
 				mon.emitGauge("hive.clustersync", 1, map[string]string{
-					"syncType": "SelectorSyncSets",
-					"name":     s.Name,
-					"result":   string(s.Result),
+					"name":    s.Name,
+					"status":  string(s.Result),
+					"type":    "SelectorSyncSets",
+					"message": cleanString(s.FailureMessage),
 				})
 				if mon.hourlyRun {
 					mon.log.WithFields(logrus.Fields{
-						"syncType":           "SelectorSyncSets",
 						"name":               s.Name,
-						"result":             string(s.Result),
+						"status":             string(s.Result),
+						"type":               "SelectorSyncSets",
 						"firstSuccessTime":   timeToString(s.FirstSuccessTime),
 						"lastTransitionTime": timeToString(&s.LastTransitionTime),
-						"failureMessage":     s.FailureMessage,
+						"message":            cleanString(s.FailureMessage),
```

Comment:
LGTM. 
Only thing needed to be verified with a Canary deployment is that the message us being populated in required column of the logs table. It did not appear with other fields before this change.